pkg/trace/api/normalizer.go

```diff
@@ -32,37 +35,74 @@ var (
 )
 
 // normalize makes sure a Span is properly initialized and encloses the minimum required info
-func normalize(s *pb.Span) error {
+func normalize(ts *info.TagStats, s *pb.Span, firstSpan *pb.Span) error {
+	if s.TraceID == 0 {
+		atomic.AddInt64(&ts.TracesDropped.TraceIDZero, 1)
+		log.Debugf("Dropping invalid trace (reason:trace_id_zero): %s", s)
+		return errors.New("trace_id_zero")
```

Comment:
A better way to do this is:
```go
return fmt.Errorf("Dropping invalid trace (reason:trace_id_zero): %s", s)
```
And not logging here. We should log where the error is returned, at the very call of `normalizeTrace`. Then, this makes the errors more useful, rather than just a filler with a random name.

Reply:
Good point, updated